c/secp256k1_keccak256_lock.h

```diff
@@ -15,531 +16,29 @@
-  ret = extract_witness_lock(temp, witness_len, &lock_bytes_seg);
-  if (ret != 0) {
-    return ERROR_ENCODING;
-  }
-
-  if (lock_bytes_seg.size != SIGNATURE_SIZE) {
-    return ERROR_ARGUMENTS_LEN;
-  }
-  memcpy(lock_bytes, lock_bytes_seg.ptr, lock_bytes_seg.size);
-
-  /* Load tx hash */
-  unsigned char tx_hash[BLAKE2B_BLOCK_SIZE];
-  len = BLAKE2B_BLOCK_SIZE;
-  ret = ckb_load_tx_hash(tx_hash, &len, 0);
-  if (ret != CKB_SUCCESS) {
-    return ret;
-  }
-  if (len != BLAKE2B_BLOCK_SIZE) {
-    return ERROR_SYSCALL;
-  }
-
-  /* Prepare sign message */
-  unsigned char message[BLAKE2B_BLOCK_SIZE];
-  SHA3_CTX sha3_ctx;
-  keccak_init(&sha3_ctx);
-  keccak_update(&sha3_ctx, tx_hash, BLAKE2B_BLOCK_SIZE);
-
-
-  /* Clear lock field to zero, then digest the first witness */
-  memset((void *)lock_bytes_seg.ptr, 0, lock_bytes_seg.size);
-  keccak_update(&sha3_ctx, (unsigned char *)&witness_len, sizeof(uint64_t));
-  keccak_update(&sha3_ctx, temp, witness_len);
-
-  /* Digest same group witnesses */
-  size_t i = 1;
-  while (1) {
-    len = MAX_WITNESS_SIZE;
-    ret = ckb_load_witness(temp, &len, 0, i, CKB_SOURCE_GROUP_INPUT);
-    if (ret == CKB_INDEX_OUT_OF_BOUND) {
-      break;
-    }
-    if (ret != CKB_SUCCESS) {
-      return ERROR_SYSCALL;
-    }
-    if (len > MAX_WITNESS_SIZE) {
-      return ERROR_WITNESS_SIZE;
-    }
-    keccak_update(&sha3_ctx, (unsigned char *)&len, sizeof(uint64_t));
-    keccak_update(&sha3_ctx, temp, len);
-    i += 1;
-  }
-  /* Digest witnesses that not covered by inputs */
-  i = calculate_inputs_len();
-  while (1) {
-    len = MAX_WITNESS_SIZE;
-    ret = ckb_load_witness(temp, &len, 0, i, CKB_SOURCE_INPUT);
-    if (ret == CKB_INDEX_OUT_OF_BOUND) {
-      break;
-    }
-    if (ret != CKB_SUCCESS) {
-      return ERROR_SYSCALL;
-    }
-    if (len > MAX_WITNESS_SIZE) {
-      return ERROR_WITNESS_SIZE;
-    }
-    keccak_update(&sha3_ctx, (unsigned char *)&len, sizeof(uint64_t));
-    keccak_update(&sha3_ctx, temp, len);
-
-    i += 1;
-  }
-  keccak_final(&sha3_ctx, message);
-
-  keccak_init(&sha3_ctx);
-  /* personal hash, ethereum prefix  \u0019Ethereum Signed Message:\n32  */
-  unsigned char eth_prefix[28]= {
-0x19, 0x45, 0x74, 0x68, 0x65, 0x72, 0x65, 0x75, 0x6d, 0x20, 0x53, 0x69 ,0x67, 0x6e, 0x65, 0x64, 0x20, 0x4d, 0x65, 0x73, 0x73, 0x61, 0x67, 0x65, 0x3a, 0x0a, 0x33, 0x32
-  };
-  keccak_update(&sha3_ctx, eth_prefix, 28);
-  keccak_update(&sha3_ctx, message, 32);
-  keccak_final(&sha3_ctx, message);
-
-  /* verify signature with peronsal hash */
-  ret = verify_signature(message, lock_bytes, eth_address);
-  if(ret == CKB_SUCCESS){
-    return CKB_SUCCESS;
-  }
-
-  /* Calculate Typed Data hash */
-  ret = calculate_typed_data(message, message);
+  ret = get_signature_from_trancation(&chain_id, message, lock_bytes);
   if(ret != CKB_SUCCESS){
     return ret;
   }
 
-  /* verify signature with typed data hash */
-  return verify_signature(message, lock_bytes, eth_address);
+  if(chain_id == 1){
```

Comment:
I saw two different coding style. Since the project is forked from an existing project, it's better to keep the original coding style, which leaves spaces around braces.